constellation/constellation.py

```diff
@@ -113,13 +116,19 @@ def start(self, prefix, network, volumes, data=None):
         nm = self.name_external(prefix)
         print("Starting {} ({})".format(self.name, str(self.image)))
         mounts = [x.to_mount(volumes) for x in self.mounts]
-        x = cl.containers.run(str(self.image), self.args, name=nm,
-                              detach=True,
-                              mounts=mounts, network="none", ports=self.ports,
-                              environment=self.environment,
-                              entrypoint=self.entrypoint,
-                              working_dir=self.working_dir,
-                              labels=self.labels)
+        cl.images.pull(str(self.image))
```

Comment:
I don't think we want this pull here, it can be quite slow and assumes network access. Ensuring that the expected containers are present is better done elsewhere

Reply:
trying to figure out exactly where to do this, shall I just do a verification in the ConstellationContainer initialiser? that feels like a natural place to do it

Reply:
```
def ensure_image(name):
    cl = docker.from_env()
    try:
        cl.images.get(name)
    except docker.errors.ImageNotFound:
        print(f"Pulling {name}")
        cl.images.pull(name)
```

perhaps use a utility like that? Or punt for now and just make sure that CI has the images we need before it runs?

Reply:
yep was gonna do that! sorry badly worded, my question was more where to stick this utility? in `__init__` of ConstellationContainer? just after we create the redis constellation containers in the tests that were failing? i think it should be in the `__init__` of constellation container because the start function assumes image is pull

Reply:
probably reasonable to put it just above any container.run call

Reply:
as discussed in standup, for now I have only put it in the tests that need redis image